tasks/staking.ts

```diff
@@ -36,53 +38,63 @@ export function load() {
   task("staking:single:pause")
     .addPositionalParam("stakingRewardAddress", "Staking reward pool contract address")
     .setAction(async ({ stakingRewardAddress }, hre) => {
-      const deployer = await getDeployer(hre);
-      const stakingRewardContract = (await hre.ethers.getContractAt("StakingRewards", stakingRewardAddress)) as StakingRewards;
-      if (!(await stakingRewardContract.connect(deployer).paused())) {
-        await (await stakingRewardContract.pause()).wait();
+      const recorder = await defaultRecorder(hre);
+
+      const stakingRewardContract = toRc((await hre.ethers.getContractAt("StakingRewards", stakingRewardAddress)) as StakingRewards, recorder);
+      if (!(await stakingRewardContract.paused())) {
+        await stakingRewardContract.record.pause();
         console.log(`StakingPool ${stakingRewardAddress} is now paused!`);
       } else {
         console.log(`StakingPool ${stakingRewardAddress} is already paused`);
       }
+      await recorder.execute();
```

Comment:
NIT
we can move the execute() to be inside the if statement as it will execute nothing in case the staking pool is already paused